cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -0,0 +1,485 @@
+use std::fs::{File, OpenOptions};
+use std::io::{BufReader, BufWriter};
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+use fslock::LockFile;
+use serde::{Deserialize, Serialize};
+use tempfile::tempdir_in;
+use tracing::debug;
+
+use super::environment::{path_hash, CanonicalPath, CanonicalizeError, EnvironmentPointer};
+use crate::data::Version;
+use crate::flox::Flox;
+use crate::utils::traceable_path;
+
+pub const ENV_REGISTRY_FILENAME: &str = "env-registry.json";
+
+/// Errors encountered while interacting with the environment registry.
+#[derive(Debug, thiserror::Error)]
+pub enum EnvRegistryError {
+    #[error("couldn't acquire environment registry file lock")]
+    AcquireLock(#[source] fslock::Error),
+    #[error("couldn't open environment registry file")]
+    OpenRegistry(#[source] std::io::Error),
+    #[error("couldn't parse environment registry")]
+    ParseRegistry(#[source] serde_json::Error),
+    #[error("couldn't read environment pointer")]
+    ReadPointer(#[source] std::io::Error),
+    #[error("couldn't parse environment pointer")]
+    ParsePointer(#[source] serde_json::Error),
+    #[error("no '.flox' directory at provided path")]
+    DotFloxDoesntExist(#[source] CanonicalizeError),
+    #[error("failed to open temporary file for registry")]
```

Comment:
Yeah this was done because it didn't make sense to have some methods return `EnvironmentError`, which would then need to be mapped to `EnvRegistryError`, which would then need to be mapped back to `EnvironmentError`. You're right that these are no longer used though. I wish there was a way to know when error enum variants weren't used.